docs/architecture/adr-009-proof-size-analysis.md

```diff
@@ -0,0 +1,155 @@
+    > - Messages begin at a location aligned with the largest power of 2 that is not larger than the message length or k.
+
+    To this: **Messages start at an index that is a multiple of its `msgMinSquareSize`.**
+
+    As an example, we have this diagram. Message 1 is three shares long and is followed by message 2, which is 11 shares long, so the `msgMinSquareSize` of the second message is equal to four. Therefore we have a padding of 5 shares shown in light blue. Furthermore, with the new non-interactive default rule set, a message of size 11 can start in this block at index zero and index three because they are multiples of four. Therefore, we save four shares of padding while retaining the same commitment.
+
+    ![Padding Savings](./assets/padding-savings.png)  
+
+### Defining variables for this Analysis
+
+- n:= Message length in number of shares
+- k:= Square size
+- r:= Number of rows a message fills in a Celestia square.
+
+### The following Questions will be answered
+
+- Why has a message in a bigger square size even with square size independent commitments O(log(n)) proof size?
+- What is the worst-case constructible message in a square to have the biggest impact from switching from O(log(n)) to O(sqrt(n))?
+- Why can we not use the same trick that we used in Question 1 in a single row for more efficient proofs over the row roots?
+- How big is the proof size for this message?
+- What is the worst constructible block with the most amount of padding with old and new non-interactive defaults?
+- Quantify padding and proof size cost.
+
+## 1. Why has a message in a bigger square size even with square size independent commitments O(log(n)) proof size?
+
+If you use the old NI-Rules then the message begins at a location aligned with the largest power of 2 that is not larger than the message length or k. Because the subtree roots are aligned you can skip some subtree roots and calculate their parents.
+In the example below instead of proving H1, H2, H3, and H4 to the DataRoot you can prove H10. **H10 is part of the commitment generation and part of the Merkle tree to the DataRoot.** That is why you can use it for more efficient proofs. In smaller square sizes, you cannot do this, because H10 does not exist. The nodes in **red** are the subtree nodes that you need to provide for the message inclusion proof. The nodes in **blue** are the additional nodes for the Merkle proof.
+
+![Efficient Merkle Proofs with ADR008](./assets/effizicient-proof-size-ADR008.png)
+
+So why can you not do it with the new NI-Rules? This is because H10 is not generated. In the diagram below the first 8 shares are in the row before and therefore the tree structure changes. The previous subtree root H10 is now H23 and cannot be used for the efficiency boost.
```

Comment:
[optional] I realize this document uses NI-Rules in many places so it may not be desirable to use the longer form "non-interactive default rules". However, the longer form avoids any ambiguity around abbreviations which we've historically had with NID (sometimes confused with namespaceID). If we want to keep the abbreviation, we may clarify its first usage (I think this instance) by including the longer form in parenthesis after: NI-Rules (non-interactive rules)

```suggestion
So why can you not do it with the new non-interactive default rules? This is because H10 is not generated. In the diagram below the first 8 shares are in the row before and therefore the tree structure changes. The previous subtree root H10 is now H23 and cannot be used for the efficiency boost.
```

Reply:
I will switch to the longer form. Maybe it's hinders readability to add new abbreviations. I did not want to use NID, because of the misunderstanding you mentioned.